faktory/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if and (.Values.metrics.enabled) (.Values.metrics.serviceMonitor.enabled) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "faktory.fullname" . }}
+  {{- if .Values.metrics.serviceMonitor.namespace }}
+  namespace: {{ .Values.metrics.serviceMonitor.namespace }}
+  {{- else }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
+  labels:
+    app: {{ template "faktory.name" . }}
+    chart: {{ template "faktory.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
```

Comment:
If I understand correctly, these labels are used so that the prometheus object can specify which ServiceMonitors should be included. Is that correct?

If so, what do you think about just naming this value `.Values.metrics.serviceMonitor.labels`? Then you could just insert them with `{{ .Values.metrics.serviceMonitor.labels | indent 4 }}` and I think `labels` describes what they are sufficiently.

Reply:
`{{ .Values.metrics.serviceMonitor.labels | indent 4 }}` doesn't work because Helm templates are trying to output something like `map[key:value]` (Go internal representation?) which apparently is not a valid YAML. Couldn't find YAML formatting funtion neither in [Helm Template guide](https://helm.sh/docs/chart_template_guide/), nor in [Sprig](https://masterminds.github.io/sprig/), nor in [Go Templates](https://godoc.org/text/template)

So just changed `selector` to `labels`

Reply:
Ah, finally I've found `toYaml` function. But what is crucial in current code is value quoting.

So if we just insert labels from Values as is with following template:
```
    {{- if .Values.metrics.serviceMonitor.labels }}
    {{- toYaml .Values.metrics.serviceMonitor.labels | nindent 4 }}
    {{- end }}
```
And place `foo: 1` for labels in Values, then following error will occur:
```
Error: UPGRADE FAILED: unable to decode "": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Namespace: Name: Labels: ReadString: expects " or n, but found 1, error found in #10 byte of ...|y","foo":1,"helm.sh/|..., bigger context ...|:"Helm","app.kubernetes.io/name":"faktory","foo":1,"helm.sh/chart":"faktory-0.11.0","release":"prome|...
```
So labels should be string and this should be ensured.